container/sagemaker/serve.py

```diff
@@ -115,6 +116,7 @@ def _read_nginx_template(self):
             return template
 
     def _start_tfs(self):
+        subprocess.check_call('tensorflow_model_server --version'.split())
```

Comment:
yes, this logs the tensorflow serving and tensorflow version info. if we log it from python, we need to capture and parse it. since all logs in-container go to stdout anyway this seems more efficient and less brittle

Reply:
My suggested approach is the same that we do for GPU [here](https://github.com/aws/sagemaker-containers/blob/master/src/sagemaker_containers/_env.py#L286):

```python
def num_gpus():  # type: () -> int
    """The number of gpus available in the current container.
    Returns:
        int: number of gpus available in the current container.
    """
    try:
        cmd = shlex.split('nvidia-smi --list-gpus')
        output = subprocess.check_output(cmd).decode('utf-8')
        return sum([1 for x in output.split('\n') if x.startswith('GPU ')])
    except (OSError, subprocess.CalledProcessError):
        logger.info('No GPUs detected (normal if no gpus installed)')
        return 0
```

Reply:
yes, this is brittle. it relies on specifics of the command's text output that may change. in your example, parsing is necessary to get a count of the gpus. here, we are just dumping some diagnostic info to the log.

Reply:
Is there a middle-ground where you get the output of the command into a string and then dump that to the log? Agree that parsing the version info from the command looks dangerous. Using logging at a level where we can is good, because it gives us a little more control over how what we log - even if it ultimately we dumpt to stdout.

Reply:
I'm open to any options you guys decide here.

Reply:
updated